eclipse/plugin/src/nl/tudelft/watchdog/eclipse/logic/ui/listeners/GeneralActivityListener.java

```diff
@@ -15,13 +15,15 @@
 
 	/** Constructor. */
 	public GeneralActivityListener(Display display) {
-		display.addFilter(SWT.KeyDown | SWT.KeyUp | SWT.MouseDown | SWT.MouseUp
-				| SWT.MouseMove, new Listener() {
-
-			@Override
-			public void handleEvent(Event event) {
-				WatchDogEventType.USER_ACTIVITY.process(event);
-			}
+		display.syncExec(() -> {
```

Comment:
Why this change?

Reply:
There is no guarantee that this listener runs on the UI Thread. This actually happens when you run in headless mode in the tests. Therefore, the method has to be invoked on that UI Thread with `display.syncExec`